src/script/common/c_content.cpp

```diff
@@ -562,24 +562,48 @@ void read_content_features(lua_State *L, ContentFeatures &f, int index)
 	if(lua_istable(L, -1)){
 		int table = lua_gettop(L);
 		lua_pushnil(L);
-		int i = 0;
+		int index = -1;
+		int max_index = -1;
 		while(lua_next(L, table) != 0){
+			if (lua_type(L, -2) == LUA_TNUMBER) {
+				index = lua_tonumber(L, -2) - 1;
+				if (index > max_index) {
+					max_index = index;
+				}
+			} else {
+				lua_pop(L, 1);
+				continue;
+			}
 			// Read tiledef from value
-			f.tiledef[i] = read_tiledef(L, -1, f.drawtype);
+			f.tiledef[index] = read_tiledef(L, -1, f.drawtype);
 			// removes value, keeps key for next iteration
 			lua_pop(L, 1);
-			i++;
-			if(i==6){
+			if(index == 5) {
 				lua_pop(L, 1);
 				break;
 			}
 		}
+		int index_limit = 6;
+		if(f.drawtype == NDT_RAILLIKE) {
+			index_limit = 4;
+		}
 		// Copy last value to all remaining textures
-		if(i >= 1){
-			TileDef lasttile = f.tiledef[i-1];
-			while(i < 6){
-				f.tiledef[i] = lasttile;
-				i++;
+		if(max_index > -1){
+			index = max_index;
+			TileDef lasttile = f.tiledef[index];
+			while(index < index_limit){
+				f.tiledef[index] = lasttile;
+				index++;
+			}
+		}
+		// Raillike: Fallback to 'straight' texture for
+		// missing 'single' and 'end' texture
+		if(f.drawtype == NDT_RAILLIKE) {
+			if(f.tiledef[4].name.empty()) {
+				f.tiledef[4] = f.tiledef[0];
+			}
+			if(f.tiledef[5].name.empty()) {
+				f.tiledef[5] = f.tiledef[0];
```

Comment:
Putting this specifically into a scriptapi feels like the wrong place.
Isn't there a function in `nodedef.cpp` that deals with compatibily like this?